### PR TITLE
xmobar: Add module

### PR DESCRIPTION
**Pull request by t4ccer**
### Description

Add `xmobar` status bar modue.

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/doc/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [x] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -321,3 +321,5 @@
 /modules/xresources.nix                               @rycee
 
 /modules/xsession.nix                                 @rycee
+
+/modules/xmobar.nix                                   @t4ccer
```
Comment (ncfavier) [inline .github/CODEOWNERS:325]: This should be `/modules/programs/xmobar.nix`, and you should add `/tests/modules/programs/xmobar`.

The list is ordered alphabetically, but you should group the tests line just after the module line.

File: modules/modules.nix [hunk 2]
```diff
@@ -150,6 +150,7 @@ let
     (loadModule ./programs/zplug.nix { })
     (loadModule ./programs/zsh.nix { })
     (loadModule ./programs/zsh/prezto.nix { })
+    (loadModule ./programs/xmobar.nix { })
```
Comment (ncfavier) [inline modules/modules.nix:153]: Alphabetically ordered too.

File: modules/programs/xmobar.nix [hunk 3]
```diff
@@ -0,0 +1,53 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let cfg = config.programs.xmobar;
+in {
+  options.programs.xmobar = {
+    enable = mkEnableOption "Xmobar: a minimalistic status bar";
```
Comment (ncfavier) [inline modules/programs/xmobar.nix:8]: I think changing the colon to a comma would read better in context ("Whether to enable Xmobar, a minimalistic status bar").

File: tests/default.nix [hunk 4]
```diff
@@ -83,6 +83,7 @@ import nmt {
     ./modules/programs/vscode
     ./modules/programs/zplug
     ./modules/programs/zsh
+    ./modules/programs/xmobar
```
Comment (ncfavier) [inline tests/default.nix:86]: Alphabetically ordered too.

**Review by ncfavier — CHANGES_REQUESTED** [review]

**Comment by t4ccer** [comment]
Reordered lists to make them alphabetically ordered, and changed colon to a comma in `enable` option description

File: .github/CODEOWNERS [hunk 5]
```diff
@@ -321,3 +324,4 @@
 /modules/xresources.nix                               @rycee
 
 /modules/xsession.nix                                 @rycee
+
```
Comment (sumnerevans) [inline .github/CODEOWNERS:327]: nit: remove extra newline

File: modules/programs/xmobar.nix [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+with lib;
+
+let cfg = config.programs.xmobar;
+in {
+  options.programs.xmobar = {
+    enable = mkEnableOption "Xmobar, a minimalistic status bar";
+
+    package = mkOption {
+      default = pkgs.haskellPackages.xmobar;
```
Comment (sumnerevans) [inline modules/programs/xmobar.nix:11]:
```suggestion
      default = pkgs.haskellPackages.xmobar;
      defaultText = literalExample "pkgs.haskellPackages.xmobar";
```

**Review by sumnerevans — CHANGES_REQUESTED** [review]
Thanks! Looks good, save a few nits!

File: modules/programs/xmobar.nix [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+              , Run StdinReader
+              ]
+          , sepChar     = "%"
+          , alignSep    = "}{"
+          , template    = "  %StdinReader% | %cpu% | %memory% | %enp3s0%  }{%date%  "
+          }
+      '';
+      type = types.lines;
+      description =
+        "Extra configuration lines to add to ~/.config/xmobar/.xmobarrc";
```
Comment (sumnerevans) [inline modules/programs/xmobar.nix:44]: sorry for not noticing this on my last review pass

```suggestion
      description = ''
        Extra configuration lines to add to
        <filename>$XDG_CONFIG_HOME/xmobar/.xmobarrc</filename>.
        See
        <link xlink:href="LINK_TO_DOCS" />
        for options.
      '';
```

**Review by sumnerevans — CHANGES_REQUESTED** [review]

**Review by sumnerevans — APPROVED** [review]
Checked the docs and it looks good. Thanks for the contribution!

**Pull request merged**